tests/v2/Controller/DocumentController/SwitchPrivateTest.php

```diff
@@ -48,4 +49,27 @@ public function testRoute(
             $this->assertRoute($newUrl, 403, $userMail, null, $method);
         }
     }
+
+    public function provideTestSwitchPrivateWithParentFolder(): ?\Generator
+    {
+        yield 'Should return 403 status code when authenticated as beneficiaire and document has parent folder' => [
+            self::URL,
+            403,
+            BeneficiaryFixture::BENEFICIARY_MAIL,
+        ];
+        yield 'Should return 403 status code when authenticated as member with relay in common and document has parent folder' => [
+            self::URL,
+            403,
+            MemberFixture::MEMBER_MAIL_WITH_RELAYS_SHARED_WITH_BENEFICIARIES,
+        ];
+    }
+
+    /** @dataProvider provideTestSwitchPrivateWithParentFolder */
+    public function testSwitchPrivateWithParentFolder(string $url, int $statusCode, string $userMail): void
+    {
+        $beneficiary = BeneficiaireFactory::findByEmail(BeneficiaryFixture::BENEFICIARY_MAIL)->object();
+        $document = DocumentFactory::findOrCreate(['beneficiaire' => $beneficiary, 'dossier' => FolderFactory::random()])->object();
+
+        $this->assertRoute(sprintf($url, $document->getId()), $statusCode, $userMail);
+    }
```

Comment:
This comment applies for all your tests : 
 It seems like we are only testing cases for wich we can not Switch to private. So I think you're test should be named `testCanNotSwitchPrivateWithParentFolder` (or even `testCanNotSwitchPrivateDocumentWithParentFolder`). 
 And you don"t have to pass `$url` or `$statuscode` as arguments if you always use the same values (this apply to any argument that always have the same value in our test cases)